include/zephyr/sys/reboot.h

```diff
@@ -20,19 +13,66 @@
 extern "C" {
 #endif
 
-#define SYS_REBOOT_WARM 0
-#define SYS_REBOOT_COLD 1
+/**
+ * @brief System reboot
+ * @defgroup sys_reboot System reboot
+ * @ingroup os_services
+ * @{
+ */
+
+/** Reboot modes. */
+enum sys_reboot_mode {
+	/**
+	 * Default reboot.
+	 *
+	 * When using this mode, caller lets the system to decide the most
+	 * appropriate reboot mode (platform dependent). This option can be
+	 * used to write portable code, where there is not enough context
+	 * information to decide the most suitable reboot mode.
+	 */
+	SYS_REBOOT_DEFAULT = -1,
+	/**
+	 * Cold reboot.
+	 *
+	 * This mode is the one preserving the least state possible in the
```

Comment:
Would it be better to say for the cold reboot something like:
"The system will be rebooted in a way comparable to a power on reset.
What this means depends on the platform, but you can normally expect the device will boot from ROM, and  its peripherals will be re-initialiazed"

And for the warm reboot:
"Typically a faster and lighter reboot than COLD. What this means and what is reset is greatly platform dependent."

Reply:
good point, added these descriptions. thanks!